src/main/java/uk/gov/companieshouse/presenter/account/interceptor/UserAuthenticationInterceptor.java

```diff
@@ -0,0 +1,144 @@
+import uk.gov.companieshouse.api.util.security.TokenPermissions;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.util.RequestLogger;
+
+import org.springframework.web.servlet.AsyncHandlerInterceptor;
+
+/**
+ * Checks each request to the service to verify that the user is authorised to
+ * use the service.
+ */
+@Component
+public class UserAuthenticationInterceptor implements AsyncHandlerInterceptor, RequestLogger {
+
+    private static final Key USER_PRESENTER = Key.USER_PRESENTER;
+    private static final String CREATE = Value.CREATE;
+
+    final Logger logger;
+    private final AuthenticationHelper authHelper;
+
+    /**
+     * Constructor which configures the logger and authentication helper class.
+     *
+     * @param authHelper the {@link AuthenticationHelper}
+     * @param logger     the specified logger
+     */
+    @Autowired
+    public UserAuthenticationInterceptor(final AuthenticationHelper authHelper, final Logger logger) {
+        this.authHelper = authHelper;
+        this.logger = logger;
+    }
+
+    /**
+     * Ensure requests are authenticated for a user.
+     */
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response,
+            Object handler) {
+
+        final String identityType = authHelper.getAuthorisedIdentityType(request);
+        boolean validIdentity = false;
+        boolean validPermission = false;
+        final String requestMethod = authHelper.getRequestMethod(request);
+        boolean isGetRequest = authHelper.isGetMethod(requestMethod);
+
+        if (requestMethod == null || requestMethod.isEmpty()) {
+            logger.debugRequest(request, "UserAuthenticationInterceptor error: no request method", null);
+        }
+
+        if (identityType == null || identityType.isEmpty()) {
+            logger.debugRequest(request, "UserAuthenticationInterceptor error: no authorised identity type", null);
+        }
+
+        final String identity = authHelper.getAuthorisedIdentity(request);
+
+        if (identity == null || identity.isEmpty()) {
+            logger.debugRequest(request, "UserAuthenticationInterceptor error: no authorised identity", null);
+        }
+
+        if (authHelper.isOauth2IdentityType(identityType)) {
+            validIdentity = validateOAuth2Identity(request, identity);
+        } else {
+            logger.debugRequest(request, "UserAuthenticationInterceptor error: identity type not oauth2",
+                    null);
+        }
+
+        if (isGetRequest) {
+            logger.debugRequest(request, "UserAuthenticationInterceptor info: Request is GET - ignore permissions.",
+                    null);
+        } else {
+            validPermission = validateUserPresenterPermission(request);
+        }
+
+        if (!validIdentity) {
+            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+            return false;
+        } else if (!isGetRequest && !validPermission) {
```

Comment:
The healthcheck doesn't have a valid identity, so won't get this far. I would probably do the is GET check at the start of the method and return true immediately if it is.